### PR TITLE
[core][devex] Move ray/util build targets to separate build files

### PR DESCRIPTION
**Pull request by vitsai**
## Why are these changes needed?

Some build file cleanup before we separate out into finer targets.

## Related issue number

 https://github.com/ray-project/ray/issues/36611

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by raulchen** [comment]
Is there a master issue that describes this effort? e.g., what problems are we going to resolve? and the execution plan. 

For this particular PR, it seems to me that the only benefit is to slim the the BUILD file. And I don't see why I can't do it in place.

**Comment by vitsai** [comment]
Created https://github.com/ray-project/ray/issues/36611 to track this.

Looks like it can be done incrementally in place after all; the mistake was putting BUILD files in the src and ray directories. Once you put a BUILD file in a directory, all targets building that subtree cannot be in a BUILD file in an ancestor of that directory.

However, it seems like there is a way to skip the `src` and `ray` directories the way the `protobuf` folder has done, perhaps by fiddling with visibility settings.

File: src/ray/util/test/BUILD [hunk 1]
```diff
@@ -0,0 +1,147 @@
+load("@rules_cc//cc:defs.bzl", "cc_test")
```
Comment (jjyao) [inline src/ray/util/test/BUILD:1]: src/ray/util/tests based on the naming convention.

**Review by jjyao — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

**Comment by pcmoritz** [comment]
Nice, it is great that you can do it without moving the source tree :)

**Pull request merged**